src/devicecontroller/DefaultDeviceController.ts

```diff
@@ -520,9 +548,24 @@ export default class DefaultDeviceController implements DeviceControllerBasedMed
         newDevice.stream = stream;
         newDevice.constraints = proposedConstraints;
       } else {
+        // TODO: remove check when the bug is fixed.
+        if (
+          kind === 'audio' &&
+          this.activeDevices[kind] &&
+          device &&
+          this.getDeviceIdStr(device) === 'default'
```

Comment:
Why does it matter what the device label of the device is (e.g. `default`)? Shouldn't this logic apply any time there is a request for a device that has the same device id as existing but a different group id?

Reply:
deviceId is unique but the groupId is not. With the 'default' in this  condition check, we are making sure that the releaseMediaStream() or stop() is called only for the the device which has the deviceId 'default'

Reply:
However if we do a check 
this.getDeviceIdStr(device) === JSON.stringify(this.activeDevices[kind].constraints.audio).deviceId.exact 
instead, the 'default' check can be avoided

Reply:
My question is really what about this workaround needs to be specific to the device that is labeled `default`? Do we need to know that the device is the default one? Can we not trigger the workaround on any existing device whose group id is now different?